File: c#/array/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> newMap = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int complement = target - nums[i];

            if (newMap.ContainsKey(complement))
            {
                return new int[] {newMap[complement], i};
            }

            newMap[nums[i]] = i;
        }

        return new int[0];
    }
}

File: python/array/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        num_map = {}
        
        for i, num in enumerate(nums):
            target_dif = target - num
            
            if target_dif in num_map:
                return [num_map[target_dif], i]
            
            num_map[num] = i
        
        return []